django/db/migrations/state.py

```diff
@@ -97,16 +97,33 @@ def __init__(self, models=None, real_apps=None):
             self.real_apps = set()
         self.is_delayed = False
         # {remote_model_key: {model_key: [(field_name, field)]}}
-        self.relations = None
+        self._relations = None
+
+    @property
+    def relations(self):
+        if self._relations is None:
+            self.resolve_fields_and_relations()
+        return self._relations
 
     def add_model(self, model_state):
-        app_label, model_name = model_state.app_label, model_state.name_lower
-        self.models[(app_label, model_name)] = model_state
+        model_key = model_state.app_label, model_state.name_lower
+        self.models[model_key] = model_state
+        if self._relations is not None:
+            concretes, _ = self._get_concrete_models_mapping_and_proxy_models()
```

Comment:
Changes in hooks are included in #14781.

Reply:
This is not resolved. You don't need to pass `concretes` anymore because `resolve_model_relations()` handles `concretes=None`. Have you seen my [comment](https://github.com/django/django/pull/14587/files#r692704479)?

Reply:
Sorry! I missed that. I am changing the same in the upcoming commit.